edit-post/index.js

```diff
@@ -57,6 +57,15 @@ export function initializeEditor( id, post, settings ) {
 	const target = document.getElementById( id );
 	const reboot = reinitializeEditor.bind( null, target, settings );
 
+	if ( 'production' !== process.env.NODE_ENV ) {
+		// Remove with 3.0 release.
```

Comment:
I wonder if we could implement this logging with an object property getter on a block's `this.props.isSelected`, so we're not feeling compelled to remove it soon, and to not log when the message is not applicable (could be confusing, since a developer might assume there's something actively wrong with a block, even if there's not).

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get

Reply:
My main concern is that often this is used as follows:
```js
isSelected && (	
	<InspectorControls key="inspector">
		...
	</InspectorControls>     
)
```

but also appears in a different context:
```js
{ images.map( ( img, index ) => (
	<li className="blocks-gallery-item" key={ img.id || img.url }>
		<GalleryImage
			url={ img.url }
			alt={ img.alt }
			id={ img.id }
			isSelected={ isSelected && this.state.selectedImage === index }
			onRemove={ this.onRemoveImage( index ) }
			onSelect={ this.onSelectImage( index ) }
			setAttributes={ ( attrs ) => this.setImageAttributes( index, attrs ) }
			caption={ img.caption }
		/>
	</li>
) ) }
```

It is very hard to handle it on a case by case basis.